File: Samples/Factory Pattern/Sample_1 - CS/LogisticsBySea.cs
namespace Sample
{
    internal class LogisticsBySea : Logistics
    {
        public override void CreateTransport()
        {
            _transport = new Ship();
        }

        public override void ExecuteDelivery()
        {
            if (_transport == null)
            {
                Console.WriteLine("Transposrt not set!");
                return;
            }

            _transport.Delivery();
        }
    }
}

File: Samples/Factory Pattern/Sample_1 - CPP/LogisticsBySea.cpp
#include "LogisticsBySea.h"

LogisticsBySea::~LogisticsBySea()
{
	delete m_delivery_transport;
}

void LogisticsBySea::CreateTransport()
{
	m_delivery_transport = new Ship();
}

void LogisticsBySea::ExecuteDelivery()
{
	if (m_delivery_transport == nullptr) {
		cout << "Transport not set!" << endl;
		return;
	}

	m_delivery_transport->Delivery();
}
